### DISCUSSION TITLE
How to auto-fitting the size of the grid

### DISCUSSION THREAD

**Discussion by FredericChang** [tailwindlabs/tailwindcss#8139] [Help]
Dear All,

devDependencies:  "tailwindcss": "^3.0.24",

Do you know how to auto-fitting the size of the grid?
I would like to create three columns as shown in Fig1.  there is a divider to separate columns into two parts.
The left side will be the navigation items, and the right side will be the contents.
![Fig2](https://user-images.githubusercontent.com/7896940/163796174-9589979e-117f-4b32-89dc-b0f896b93c6f.jpg)

However I tried to use "grid grid-cols" to separate two sides, and it works. but there're two functions I need your help with:
01. How to add a divider between two columns?
02. How to do the auto-fitting of the left column? because the size of box of <ul> is over-sized. how to limit it?

![Fig1](https://user-images.githubusercontent.com/7896940/163796175-14f81b44-23c9-41b2-a313-157ab6879e19.jpg)

```html

render(){
    return(
            <>

                <div class="grid grid-cols-8 gap-4 h-auto w-auto border-2 border-yellow-600">
                    <div class="col-start-1 col-end-3 border-4 border-red-600 h-96">
                            <div className='border-4 border-red-600 w-48 center'>
                                <ul className="border-2 border-lime-500 space-y-12">
                                    <li className="text-center center border-2 border-yellow-600">
                                        <a href=''>VACCINATIONS</a>
                                    </li>
                                    <li className="text-center center border-2 border-yellow-600">Hospital</li>
                                    <li className="text-center center border-2 border-yellow-600">Infections</li>
                                    <li className="text-center center border-2 border-yellow-600">Death Rates</li>
                                    <li className="text-center center border-2 border-yellow-600">Compliance </li>
                                    <li className="text-center center border-2 border-yellow-600">Other </li>
                                    <li className="text-center center border-2 border-yellow-600">Useful links </li>

                                </ul>
                            </div>
                    </div>
                    <div class="col-end-8 col-span-5 border-4 border-red-600">03</div>
                </div>
            </>      
    )
}

```

**Comment by horuskol** [comment]
> How to add a divider between two columns?

I assume you mean a border rather than a gap? I'd run a border down the left hand side of your main content div (03).

> How to do the auto-fitting of the left column? because the size of box of
is over-sized. how to limit it?

`grid-cols-n` is a nice way to quickly get a grid up, but since it creates `n` equal columns it can be a bit limiting.

There's a couple of options to do what you want - you can define a utility in `tailwind.config.js`:

```
  theme: {
    extend: {
      gridTemplateColumns: {
        'layout': 'auto 1fr',
      },
    },
  },
```

which you can then use as `<div class="grid grid-cols-layout">`.

Or you can use a dynamic class if you're using JIT: `<div class="grid grid-cols-[auto_1fr]">`

**Comment by Emmitt121** [comment]
Good one, I had no idea about this trick. Make sense if it works pretty well. Thanks!

[MyMileStoneCard Login
](https://www.mymilestonecard.net/)
